### PR TITLE
Support running without a nightly toolchain but still with sanitizers

### PR DESCRIPTION
**Pull request by Ekleog**
This sets `RUSTC_BOOTSTRAP` when explicitly requested, thus allowing the user to fuzz with sanitizers without a nightly toochain installed locally.

This is as per https://github.com/camshaft/bolero/issues/71#issuecomment-1126289896 ; hopefully it's clean enough to get landed :)

### CODE REVIEW

File: cargo-bolero/src/project.rs [hunk 1]
```diff
@@ -48,18 +48,26 @@ pub struct Project {
     /// Build the standard library with the provided configuration
     #[structopt(long)]
     build_std: bool,
+
+    /// Fake using a nightly toolchain while using the default toolchain by using RUSTC_BOOTSTRAP
+    #[structopt(long)]
+    fake_nightly_toolchain: bool,
```
Comment (camshaft) [inline cargo-bolero/src/project.rs:54]: Why not name this the same as what we're doing so it's clear and we don't have to document the mapping?

```suggestion
    rustc_bootstrap: bool,
```

Comment (Ekleog) [reply cargo-bolero/src/project.rs:54]: Just checking if you have opinions on this naming? I can without any issues rename the flag to `--rustc-bootstrap` in this PR, but don't know how you feel about my explanation above 😅

Comment (camshaft) [reply cargo-bolero/src/project.rs:54]: I think I still prefer to go with rustc bootstrap. It's already [public](https://github.com/rust-lang/cargo/issues/6627) and I would prefer to inherit any available documentation about what it does. Make sense?

Comment (Ekleog) [reply cargo-bolero/src/project.rs]: The link you sent is to an issue that was actually closed as "won't do without an RFC" :sweat_smile: 

That said I have no strong opinions here, so I've just force-pushed a rename to `rustc_bootstrap` :)

**Review by camshaft — APPROVED** [review]

**Pull request merged**